sequence.go

```diff
@@ -654,28 +659,36 @@ func findSequencesOnDisk(path string, opts ...FileOption) (FileSequences, error)
 
 		frame, _ := strconv.Atoi(match[2])
 		key := [2]string{match[1], match[3]}
-		frames, ok := seqs[key]
+		seq, ok := seqs[key]
+		frameWidth := len(match[2])
 		if !ok {
-			frames = []int{frame}
-			padMap[key] = padder.PaddingChars(len(match[2]))
+			seq = seqInfo{
+				Frames:   []int{frame},
+				Padding:  padder.PaddingChars(frameWidth),
+				MinWidth: frameWidth,
+			}
 		} else {
-			frames = append(frames, frame)
+			seq.Frames = append(seq.Frames, frame)
+			if seq.MinWidth > frameWidth {
+				seq.MinWidth = frameWidth
+				seq.Padding = padder.PaddingChars(frameWidth)
+			}
 		}
-		seqs[key] = frames
+		seqs[key] = seq
```

Comment:
Oh,  smart. Ya you would need the ok check just for that part. But you
still wouldn't have to create a second copy of the struct since you always
get a copy when you read from the map.

On Sun, 12 Jun 2016 9:34 PM rlei <notifications@github.com> wrote:

> In sequence.go
> <https://github.com/justinfx/gofileseq/pull/8#discussion_r66718557>:
>
> >  		}
> > -		seqs[key] = frames
> > +		seqs[key] = seq
>
> Hmm, I think there will be a problem because MinWidth will be initialized
> as 0 right? Then it would never > frameWidth
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/justinfx/gofileseq/pull/8/files/70b4433312f76bd8c05a3e169b74534bf251d583#r66718557>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AARPxlPFk7MFWehdLyRO3FC9XSSBVcz_ks5qK9KdgaJpZM4Izf7s>
> .
>